### PR TITLE
Implement ChatWorkshop/user/unreadMessagesCount endpoint

### PR DESCRIPTION
**Pull request by olsavchenk**
Implemented ChatWorkshop/user/unreadMessagesCount(https://github.com/ita-social-projects/OoS-Frontend/issues/2294)
Fixed ChatWorkshop/parent/chatrooms endpoint

### CODE REVIEW

File: OutOfSchool/OutOfSchool.WebApi.Tests/Services/ChatRoomWorkshopServiceTests.cs [hunk 1]
```diff
@@ -448,6 +448,73 @@ public async Task GetUniqueChatRoom_WhenRoomDoesNotExist_ShouldReturnNull()
...
+            .Setup(x => x.GetByParentIdAsync(parentId, It.IsAny<bool>()))
+            .ReturnsAsync(chatRooms);
+
+        // Act
+        var result = await roomService.GetCurrentUserUnreadMessagesCountAsync(parentId, role).ConfigureAwait(false);
+
+        // Assert
+        Assert.IsInstanceOf<int>(result);
+        Assert.IsNotNull(result);
+        Assert.AreEqual(2, result);
```
Comment (VadymLevkovskyi) [inline OutOfSchool/OutOfSchool.WebApi.Tests/Services/ChatRoomWorkshopServiceTests.cs]: (Code style) `2` is a magic number. Don't use them! 
For this specific case you could instead use 
`var expectedCount = chatRooms.Count(r => r.NotReadByCurrentUserMessagesCount > 0)`.
Same in test below.

File: OutOfSchool/OutOfSchool.WebApi.Tests/Services/ChatRoomWorkshopServiceTests.cs [hunk 2]
```diff
@@ -448,6 +448,73 @@ public async Task GetUniqueChatRoom_WhenRoomDoesNotExist_ShouldReturnNull()
...
+
+        roomWithSpecialModelRepositoryMock
+            .Setup(x => x.GetByParentIdAsync(parentId, It.IsAny<bool>()))
+            .ReturnsAsync(chatRooms);
+
+        // Act
+        var result = await roomService.GetCurrentUserUnreadMessagesCountAsync(parentId, role).ConfigureAwait(false);
+
+        // Assert
+        Assert.IsInstanceOf<int>(result);
```
Comment (VadymLevkovskyi) [inline OutOfSchool/OutOfSchool.WebApi.Tests/Services/ChatRoomWorkshopServiceTests.cs]: (Excessive code) Please, don't test the compiler. If code changes to other return type you'll see compiler error (where it is valuable) earlier than this unit test will fail.

File: OutOfSchool/OutOfSchool.WebApi/Services/ChatRoomWorkshopService.cs [hunk 3]
```diff
@@ -432,6 +434,22 @@ public async Task<ChatRoomWorkshopDto> GetUniqueChatRoomAsync(Guid workshopId, G
...
+        {
+            chatrooms = await GetByParentIdAsync(parentOrProviderId);
+        }
+
+        if (userRole == Role.Provider)
+        {
+            chatrooms = await GetByProviderIdAsync(parentOrProviderId);
+        }
+
+        return chatrooms.Count(chatroom => chatroom.NotReadByCurrentUserMessagesCount != 0);
```
Comment (VadymLevkovskyi) [inline OutOfSchool/OutOfSchool.WebApi/Services/ChatRoomWorkshopService.cs:446]: (Possible NRE) 
If `userRole` is neither `Role.Parent` nor `Role.Provider` - you'll get null set to `chatrooms` and thus `NullReferenceException` on `.Count()`. 
If it is a part of requirement (I'll left out of discussion downsides of the exeption-driven control flow approach), then please state it directly in code.
I.e. 
```
    public async Task<int> GetCurrentUserUnreadMessagesCountAsync(Guid parentOrProviderId, Role userRole)
    {
        var chatrooms = userRole switch {
            Role.Parent => await GetByParentIdAsync(parentOrProviderId),
            Role.Provider => await GetByProviderIdAsync(parentOrProviderId),
            _ => throw new ArgumentOutOfRangeException(nameof(userRole)),
        };

        return chatrooms.Count(chatroom => chatroom.NotReadByCurrentUserMessagesCount != 0);
    }
```

File: OutOfSchool/OutOfSchool.WebApi.Tests/Services/ChatRoomWorkshopServiceTests.cs [hunk 4]
```diff
@@ -448,6 +448,75 @@ public async Task GetUniqueChatRoom_WhenRoomDoesNotExist_ShouldReturnNull()
...
+
+        roomWithSpecialModelRepositoryMock
+            .Setup(x => x.GetByParentIdAsync(parentId, It.IsAny<bool>()))
+            .ReturnsAsync(chatRooms);
+
+        // Act
+        var result = await roomService.GetCurrentUserUnreadMessagesCountAsync(parentId, role).ConfigureAwait(false);
+
+        // Assert
+        Assert.IsNotNull(result);
```
Comment (VadymLevkovskyi) [inline OutOfSchool/OutOfSchool.WebApi.Tests/Services/ChatRoomWorkshopServiceTests.cs]: What does `Assert.IsNotNull(result);` when `result` is of `int` type? Could `int` be `null`? Why are you checking it?

Comment (olsavchenk) [reply OutOfSchool/OutOfSchool.WebApi.Tests/Services/ChatRoomWorkshopServiceTests.cs]: Understood and fixed

**Review by VadymLevkovskyi — APPROVED** [review]

File: OutOfSchool/OutOfSchool.WebApi.Tests/Services/ChatRoomWorkshopServiceTests.cs (hunk 4 above)
Comment (VadymLevkovskyi) [reply OutOfSchool/OutOfSchool.WebApi.Tests/Services/ChatRoomWorkshopServiceTests.cs]: Thank you

**Pull request merged**